cmake/configs/base.cmake

```diff
@@ -60,6 +60,12 @@ config_bool(
     DEFAULT OFF
```

Comment:
Though not directly related to changes in this PR, Autoconf seems to have a default "yes" for the enable-static config option.

Reply:
Yep, I intentionally defaulted it to `OFF` as for a lot of cases we don't need build both static and shared builds by default i.e. testing and development. I wanted to keep the default setting to just a single library artifact, keeping build times short and avoiding unnecessary compilation where possible.

Reply:
Sounds reasonable, thanks.